src/users/amoc_users_sup.erl

```diff
@@ -76,14 +81,16 @@ count_no_of_users() ->
     #storage{user_count = Atomics} = persistent_term:get(?MODULE),
     atomics:get(Atomics, 1).
 
--spec incr_no_of_users(non_neg_integer()) -> any().
-incr_no_of_users(SupNum) when SupNum > 1 ->
+-spec handle_up_user(non_neg_integer(), pid(), amoc_scenario:user_id()) -> any().
+handle_up_user(SupNum, Pid, Id) when SupNum > 1 ->
+    ets:insert(?TABLE, {Pid, Id}),
```

Comment:
can we just put `user_id` into the process dictionary of the user process? and add an interface at `amoc_user` to get `user_id` by `pid` (must be an rpc call for remote processes since process_info interface works only for the local processes)

Reply:
I'm not sure to understand what does that give. This is called by the supervisor worker, not the user process 🤔